src/py/flwr/server/grpc_server/grpc_server.py

```diff
@@ -12,45 +12,100 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Implements utility function to create a grpc server."""
+"""Implements utility function to create a gRPC server."""
+
+
 import concurrent.futures
+import sys
+from logging import ERROR
+from typing import ByteString, Optional, Tuple, Union
 
 import grpc
 
 from flwr.common import GRPC_MAX_MESSAGE_LENGTH
+from flwr.common.logger import log
 from flwr.proto import transport_pb2_grpc
 from flwr.server.client_manager import ClientManager
 from flwr.server.grpc_server import flower_service_servicer as fss
 
+FILELIKE = Union[str, bytes]
+SSLFILES = Tuple[FILELIKE, FILELIKE, FILELIKE]
+
+INVALID_SSL_FILES_ERR_MSG = """
+    When setting any of root_certificate, certificate, or private_key,
+    you have to set all of them.
+"""
+
+
+def read_to_byte_string(file_like: FILELIKE) -> ByteString:
+    """Read file_like and return as ByteString."""
+    with open(file_like, "rb") as file:
+        return file.read()
+
+
+def valid_ssl_files(ssl_files: SSLFILES) -> bool:
+    """Validate type SSLFILES and exit if invalid."""
+    is_valid = all(ssl_files)
```

Comment:
This doesn't really test if all three are set (`all()` returns `True` for empty iterators). We could perform the following checks:
- Is `ssl_files` of length `3`
- Is each element a valid path
- Does each path point to a file (not a directory)